core/src/program/blocks/span_block.rs

```diff
@@ -340,6 +334,23 @@ fn batch_ops(ops: Vec<Operation>) -> (Vec<OpBatch>, Digest) {
     (batches, hash)
 }
 
+/// Returns a CodeBlock [Span] from sequence of Span blocks provided as input.
+pub fn combine_spans(span_vec: &mut Vec<Span>) -> CodeBlock {
+    // This is to avoid redoing the same exact compution already done earlier when
+    // creating a SPAN block of operational batches from exported procedure body.
+    if span_vec.len() == 1 {
+        return CodeBlock::new_codeblock_from_span(span_vec.remove(0));
+    }
+
+    let mut ops = Vec::<Operation>::new();
+    for span in (*span_vec).iter() {
+        for batch in &span.op_batches {
+            ops.extend_from_slice(&batch.ops);
+        }
+    }
```

Comment:
Is there a reason not to call `span_vec.clear()` after the loop? Seems like `combine_spans()` is always followed by calling `clear()` on the vector.

Reply:
No specific reason as such! Actioned on it in the latest PR. Please see [2dfadc4](https://github.com/maticnetwork/miden/pull/250/commits/2dfadc435bfa0a372ce592bad43b9a047bf35c8f).